flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManager.java

```diff
@@ -414,10 +422,105 @@ protected static String getManuallyRefreshStatement(
         return insertStatement.toString();
     }
 
-    private static String stopJobWithSavepoint(
-            OperationExecutor executor,
+    private static ResultFetcher callDropMaterializedTableOperation(
+            OperationExecutor operationExecutor,
+            OperationHandle handle,
+            DropMaterializedTableOperation dropMaterializedTableOperation) {
+        ObjectIdentifier tableIdentifier = dropMaterializedTableOperation.getTableIdentifier();
+        if (operationExecutor
```

Comment:
I think the better implementation should be following code
```
        boolean tableExists = operationExecutor.tableExists(tableIdentifier);
        if (!tableExists) {
            if (dropMaterializedTableOperation.isIfExists()) {
                LOG.info(
                        "Materialized table {} does not exist, skip the drop operation.",
                        tableIdentifier);
                return ResultFetcher.fromTableResult(handle, TABLE_RESULT_OK, false);
            } else {
                throw new ValidationException(
                        String.format(
                                "Materialized table %s does not exist.",
                                tableIdentifier));
            }
        }
```

We can add a `tableExists` method for OperationExecutor:
```
    public boolean tableExists(ObjectIdentifier tableIdentifier) {
        Optional<Catalog> currentCatalog =
                sessionContext
                        .getSessionState()
                        .catalogManager
                        .getCatalog(tableIdentifier.getCatalogName());
        if (currentCatalog.isPresent()) {
            return currentCatalog.get().tableExists(tableIdentifier.toObjectPath());
        }
        return false;
    }
```